src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -0,0 +1,497 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiButton,
+  EuiFieldPassword,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiPageContent,
+  EuiSelectable,
+  EuiSpacer,
+  EuiSuperSelect,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage } from '@osd/i18n/react';
+import {
+  CreateDataSourceFormType,
+  CreateNewCredentialType,
+  CredentialsComboBoxItem,
+  CredentialSourceType,
+  DataSourceManagementContextValue,
+  ToastMessageItem,
+} from '../../../../types';
+import { Header } from '../header';
+import { getExistingCredentials } from '../../../utils';
+import { context as contextType } from '../../../../../../opensearch_dashboards_react/public';
+import {
+  CreateEditDataSourceValidation,
+  defaultValidation,
+  performDataSourceFormValidation,
+} from '../../../validation/datasource_form_validation';
+
+export interface CreateDataSourceProps {
+  handleSubmit: (formValues: CreateDataSourceFormType) => void;
+  displayToastMessage: (msg: ToastMessageItem) => void;
+  displayLoadingMask: (show: boolean) => void;
+}
+export interface CreateDataSourceState {
+  formErrors: string[];
+  formErrorsByField: CreateEditDataSourceValidation;
+  dataSourceTitle: string;
+  dataSourceDescription: string;
+  endpoint: string;
+  showCreateCredentialModal: boolean;
+  selectedCredentialSourceType: string;
+  selectedCredentials: CredentialsComboBoxItem[];
+  availableCredentials: CredentialsComboBoxItem[];
+  createCredential: CreateNewCredentialType;
+}
+
+const defaultCreateCredentialsForm: CreateNewCredentialType = {
+  title: '',
+  description: '',
+  credentialMaterials: {
+    credentialMaterialsType: 'username_password',
+    credentialMaterialsContent: {
+      username: '',
+      password: '',
+    },
+  },
+};
+
+const credentialSourceOptions = [
+  { value: CredentialSourceType.CreateCredential, inputDisplay: 'Create credential' },
```

Comment:
inputDisplay seems need localization supported

Reply:
As there are design changes, I will be doing localization in the next iteration